### ISSUE TITLE
buySubscription() returns error in production only (but user is successfully subscribed)

### ISSUE THREAD

**Issue by bethshook** [dooboolab/react-native-iap#743]
### Version of react-native-iap
^2.4.6

### Version of react-native
0.59.8

### Platforms you faced the error (IOS or Android or both?)
iOS

### Expected behavior
buySubscription method should return a purchase if payment method is valid

### Actual behavior
buySubscription returns the error "Error: Cannot connect to iTunes Store" but subscription successfully goes through. So the user is subscribed but the app does not recognize them as subscribed.

### Tested environment (Emulator? Real Device?)
Real device

### Steps to reproduce the behavior
Download the app from iTunes store. Open app and click "Subscribe" button in subscription modal. Attempt to access subscription-only content in app.

This is a new issue as the method has worked well in the past. The subscription appears as active in the user's App Store account, but the method returns the error.

**Comment by hyochan** [comment]
Yeah. I've written a [blog](https://github.com/dooboolab/react-native-iap#news-on-major-releases) on this where there may be a problem. It looks like a similar one. You need to upgrade to `v3` in this case.
